lib/viewmodels/user_posts_view_model.dart

```diff
@@ -19,7 +20,14 @@ class UserPostsViewModel extends AutoDisposeAsyncNotifier<UserPostsState> {
     final user = ref.watch(validLoggedInUserIdProvider);
 
     final postsFirestore = await postRepository.getUserPosts(user);
-    final posts = postsFirestore.map((post) {
+
+    // Sort the posts by publication time from lattest to oldest
+    final sortedPostsFirestore = postsFirestore.toList().sorted(
+          (postA, postB) =>
+              postB.data.publicationTime.compareTo(postA.data.publicationTime),
```

Comment:
What happens if two posts have the same publication ?
For when we will have millions of users x)

Reply:
Good point !

Actually, if two posts have the exact same publication time, the `compareTo` function will return 0.
Thus `postA before postB` and `postB before postA` are both valid ordering.
So, in such cases, multiple valid ordering are possibles.

Note that in practice however, because those posts all belong to the same user, this would mean that the user was able to publish 2 posts within the same microsecond which is unlikely (unless as an adversary behavior).

That was a good question thanks !

Reply:
Oh yes, that makes sense, thanks for the reply!